frontend/packages/core-api-client/src/lib/client/Database.ts

```diff
@@ -0,0 +1,31 @@
+import {
+  DatabaseCreateRequest,
+  DatabaseCreateResponse,
+  DatabaseListResponse,
+} from '../types/client/Database';
+
+import { BaseRESTClient } from './BaseRESTClient';
+
+export class DatabaseClient {
+  restClient: BaseRESTClient;
+
+  constructor(restClient: BaseRESTClient) {
+    this.restClient = restClient;
+  }
+
+  create = (
+    request: DatabaseCreateRequest,
+  ): Promise<DatabaseCreateResponse> => {
+    return this.restClient.do(
+      request,
+      '/databases',
+      'post',
+      request.object,
+      200,
+    );
+  };
+
+  list = (request: {} | undefined): Promise<DatabaseListResponse> => {
```

Comment:
what do you think about having generic list/create/... functions on the BaseRESTClient? asking because they are mostly the same. could be overkill though?

Reply:
I think maybe instead of having generic `list`/`get` etc, we could have methods for `get`/`post` as they roughly translate to the crud methods but are a bit more reusable.

Reply:
@neb42 did you mean you would do that in this PR or was that a future plan? i.e. are you still working on this?

Reply:
I think in the future, in the interest of getting this merged.